tests/integration/integration_test.py

```diff
@@ -1227,6 +1216,7 @@ def verify_avro_explicit_read_schema():
         conf = copy(cons_conf)
         if schema_registry_url:
             conf['schema.registry.url'] = schema_registry_url
+            conf['enable.partition.eof'] = True
```

Comment:
the check for schema_registry_url should be much earlier, and there's also one for the producer.
If schema_registry_url is always set (otherwise an error) then the ifs are not needed

Reply:
Right this test was added with the explicit read PR. It would appear I didn't not review the integration tests well enough. In order to keep the commit history cleaner can I refactor the integration test in a subsequent PR.

Reply:
Suggest you fix it in this PR, an additional commit for a cleanup is fine.

Reply:
ack